libraries/lib-ipc/internal/socket_guard.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <cassert>
+#include "ipc-types.h"
+
+class socket_guard
```

Comment:
A brief class comment a out the purpose?  But not hard to learn from reading it.